dqn_main.cpp

```diff
@@ -225,25 +199,31 @@ int main(int argc, char** argv) {
   std::vector<int> legal_actions(dqn::kOutputCount);
   std::iota(legal_actions.begin(), legal_actions.end(), 0);
 
-  CHECK(FLAGS_snapshot.empty() || FLAGS_weights.empty())
+  CHECK(FLAGS_critic_snapshot.empty() || FLAGS_actor_snapshot.empty()
+        || FLAGS_actor_weights.empty() || FLAGS_critic_weights.empty())
       << "Give a snapshot to resume training or weights to finetune "
       "but not both.";
 
   // Construct the solver
-  caffe::SolverParameter solver_param;
-  caffe::ReadProtoFromTextFileOrDie(FLAGS_solver, &solver_param);
-  solver_param.set_snapshot_prefix(save_path.c_str());
+  caffe::SolverParameter actor_solver_param;
+  caffe::SolverParameter critic_solver_param;
+  caffe::ReadProtoFromTextFileOrDie(FLAGS_actor_solver, &actor_solver_param);
+  caffe::ReadProtoFromTextFileOrDie(FLAGS_critic_solver, &critic_solver_param);
+  actor_solver_param.set_snapshot_prefix(save_path.c_str());
+  critic_solver_param.set_snapshot_prefix(save_path.c_str());
```

Comment:
These two need to snapshot to different paths. Otherwise they will overwrite each other.